service/src/main/java/com/kodemon/service/implementations/TrainerFightServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.kodemon.service.implementations;
+
+import com.kodemon.persistence.dao.TrainerFightDao;
+import com.kodemon.persistence.entity.Gym;
+import com.kodemon.persistence.entity.Pokemon;
+import com.kodemon.persistence.entity.Trainer;
+import com.kodemon.persistence.entity.TrainerFight;
+import com.kodemon.service.interfaces.PokemonFightService;
+import com.kodemon.service.interfaces.TrainerFightService;
+import com.kodemon.service.util.Pair;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Oliver Roch
+ */
+
+@Service
+public class TrainerFightServiceImpl implements TrainerFightService {
+    @Inject
+    private TrainerFightDao trainerFightDao;
+
+    @Inject
+    private PokemonFightService pokemonFightService;
+
+    @Override
+    public boolean wasFightForBadgeSuccessful(Trainer challenger, Trainer defender) {
+        double challengerScore = 0;
+        double defenderScore = 0;
+
+        for(int i = 0; i < 6; i++) {
+            Pokemon currentChallengerPokemon = challenger.getPokemons().get(i);
```

Comment:
What if challenger does not have 6 Pokemon? this will throw `IndexOutOfBoundsException`